test/Dialect/FIRRTL/expand-whens.mlir

```diff
@@ -589,6 +589,20 @@ firrtl.module @WhenInGroup(in %cond : !firrtl.uint<1>) {
   }
 }
 
+// Check that expand whens works for layers under when's.
+firrtl.layer @Layer bind {}
+// CHECK-LABEL: firrtl.module @LayerUnderWhen(
+// CHECK-NOT:   firrtl.when
+// CHECK: firrtl.printf %clock, %cond
```

Comment:
Can this also check that the `printf` remains under the layerblock?

Reply:
You got it, done!